sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/KnownMethods.cs

```diff
@@ -1,255 +1,255 @@
+                methodSymbol.ReturnType.Is(KnownType.System_String);
+        }
+
+        public static bool IsIDisposableDispose(this IMethodSymbol methodSymbol)
+        {
+            const string explicitName = "System.IDisposable.Dispose";
+            return methodSymbol != null &&
+                methodSymbol.ReturnsVoid &&
+                methodSymbol.Parameters.Length == 0 &&
+                (methodSymbol.Name == nameof(IDisposable.Dispose) ||
+                 methodSymbol.Name == explicitName);
+        }
+
+        public static bool IsIEquatableEquals(this IMethodSymbol methodSymbol)
+        {
+            const string explicitName = "System.IEquatable.Equals";
+            return methodSymbol != null &&
+                methodSymbol.Parameters.Length == 1 &&
+                methodSymbol.ReturnType.Is(KnownType.System_Boolean) &&
+                (methodSymbol.Name == nameof(object.Equals) ||
+                methodSymbol.Name == explicitName);
+        }
+
+        public static bool IsGetObjectData(this IMethodSymbol methodSymbol)
+        {
+            const string explicitName = "System.Runtime.Serialization.ISerializable.GetObjectData";
+            return methodSymbol != null &&
+                methodSymbol.Parameters.Length == 2 &&
+                methodSymbol.Parameters[0].Type.Is(KnownType.System_Runtime_Serialization_SerializationInfo) &&
+                methodSymbol.Parameters[1].Type.Is(KnownType.System_Runtime_Serialization_StreamingContext) &&
+                methodSymbol.ReturnsVoid &&
+                (methodSymbol.Name == "GetObjectData" ||
+                methodSymbol.Name == explicitName);
+        }
+
+        public static bool IsSerializationConstructor(this IMethodSymbol methodSymbol)
+        {
+            return methodSymbol != null &&
+                methodSymbol.MethodKind == MethodKind.Constructor &&
+                methodSymbol.Parameters.Length == 2 &&
+                methodSymbol.Parameters[0].Type.Is(KnownType.System_Runtime_Serialization_SerializationInfo) &&
+                methodSymbol.Parameters[1].Type.Is(KnownType.System_Runtime_Serialization_StreamingContext);
+        }
+
+        public static bool IsArrayClone(this IMethodSymbol methodSymbol)
+        {
+            return methodSymbol != null &&
+                methodSymbol.MethodKind == MethodKind.Ordinary &&
+                methodSymbol.Parameters.Length == 0 &&
+                methodSymbol.Name == nameof(Array.Clone) &&
+                methodSymbol.ContainingType.Is(KnownType.System_Array);
+        }
+
+        public static bool IsGcSuppressFinalize(this IMethodSymbol methodSymbol)
+        {
+            return methodSymbol != null &&
+                methodSymbol.Parameters.Length == 1 &&
+                methodSymbol.Name == nameof(GC.SuppressFinalize) &&
+                methodSymbol.ContainingType.Is(KnownType.System_GC);
+        }
+
+        public static bool IsDebugAssert(this IMethodSymbol methodSymbol)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Consider using more general type 'Microsoft.CodeAnalysis.ISymbol' instead of 'Microsoft.CodeAnalysis.IMethodSymbol'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS3242)